### PR TITLE
Add validator on : check id, route name, coordinates; Factorization of issues creation

### PR DESCRIPTION
**Pull request by BaptisteB-A**
Added id validation for routes, stops, trips and calendar.
Added coordinates.rs : verifies that the coordinates of stops (latitude & longitude) are valid and not missing.
The creation of an issue is now easier with two new constructors.

Note : two data sets have been added to avoid side effects
Added details to Issue constructor to get more precise information about the error

### CODE REVIEW

File: src/validators/check_id.rs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+                    severity: Severity::Error,
+                    issue_type: IssueType::MissingId,
+                    object_id: calendar.id.to_owned(),
+                    object_name: Some(format!("{}", calendar)),
+                    related_object_id: None,
+                    details: None,
+                });
+    let s = gtfs.stops
+                .values()
+                .filter(|&stop| !has_id(&**stop))
```
Comment (antoine-de) [inline src/validators/check_id.rs]: can't you just use `*` ?

File: src/validators/check_id.rs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    let gtfs = gtfs_structures::Gtfs::new("test_data/check_id").unwrap();
+    let issues = validate(&gtfs);
+    let stop_id_issues: Vec<_> = issues.iter()
+        .filter(|issue| issue.object_name == Some("Null Island".to_string()))
+        .collect();
+    
+    assert_eq!(1, stop_id_issues.len());
+    assert_eq!("Null Island", stop_id_issues[0].object_name.as_ref().unwrap());
+    assert_eq!(IssueType::MissingId, stop_id_issues[0].issue_type);
+}
```
Comment (antoine-de) [inline src/validators/check_id.rs]: missing newline

File: src/validators/coordinates.rs [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    let gtfs = gtfs_structures::Gtfs::new("test_data/coordinates").unwrap();
+    let issues = validate(&gtfs);
+    let invalid_coord_issue: Vec<_> = issues.iter()
+        .filter(|issue| issue.issue_type == IssueType::InvalidCoordinates)
+        .collect();
+
+    assert_eq!(1, invalid_coord_issue.len());
+    assert_eq!("PARENT", invalid_coord_issue[0].object_id);
+    assert_eq!(IssueType::InvalidCoordinates, invalid_coord_issue[0].issue_type);
+}
```
Comment (antoine-de) [inline src/validators/coordinates.rs]: missing newline

File: src/validators/check_id.rs [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+                    issue_type: IssueType::MissingId,
+                    object_id: calendar.id.to_owned(),
+                    object_name: Some(format!("{}", calendar)),
+                    related_object_id: None,
+                    details: None,
+                });
+    let s = gtfs.stops
+                .values()
+                .filter(|&stop| !has_id(&**stop))
+                .map(|stop| Issue {
```
Comment (antoine-de) [inline src/validators/check_id.rs]: to reduce the code duplication: maybe we can do:

```rust
fn make_missing_id_issue<T: gtfs_structures::Id + Display>(o: T) -> Issue {
            Issue {
                    severity: Severity::Error,
                    issue_type: IssueType::MissingId,
                    object_id: o.id.to_owned(),
                    object_name: Some(format!("{}", o)),
                    related_object_id: None,
                    details: None,
                }
}
```

and maybe make it more generic to take the severity + issueType ?

not tested though :stuck_out_tongue_winking_eye:

File: src/validators/duration_distance.rs [hunk 5]
```diff
@@ -77,6 +80,7 @@ fn validate_speeds(
                     object_id: departure.stop.id.to_owned(),
                     object_name: Some(format!("Trip: {}", trip_id)),
```
Comment (antoine-de) [inline src/validators/duration_distance.rs]: maybe we can now use the Display here ?

**Review by antoine-de — APPROVED** [review]

**Comment by BaptisteB-A** [comment]
Added id validation for routes, stops, trips and calendar.
Added coordinates.rs : verifies that the coordinates of stops (latitude & longitude) are valid and not missing.
The creation of an issue is now easier with two new constructors.

Note : two data sets have been added to avoid side effects
Added details to Issue constructor to get more precise information about the error

**Pull request merged**